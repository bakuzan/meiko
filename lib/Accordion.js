import classNames from 'classnames';
import React, { useState, useEffect } from 'react';

import { usePrevious } from './hooks/usePrevious';
import styles from './styles/Accordion';

function AccordionToggle({ className, heading, onToggle }) {
  return (
    <button type="button" className={className} onClick={onToggle}>
      {heading}
    </button>
  );
}

function Accordion({
  children,
  className,
  contentProps = {},
  defaultIsCollapsed,
  headingProps = {},
  heading,
  toggleComponent = AccordionToggle,
  onToggle,
  ...pass
}) {
  const [isCollapsed, setCollapsed] = useState(defaultIsCollapsed ?? true);
  const prevCollapsed = usePrevious(isCollapsed);

  const Toggler = toggleComponent;

  useEffect(() => {
    const hasPrev = prevCollapsed !== undefined;
    const hasNew = prevCollapsed !== isCollapsed;

    if (onToggle && hasPrev && hasNew) {
      onToggle(isCollapsed);
    }
  }, [onToggle, isCollapsed, prevCollapsed]);

  return (
    <div
      {...pass}
      className={classNames('accordion', styles.accordion, className)}
    >
      <div
        {...headingProps}
        className={classNames(
          'accordion__heading',
          styles.accordion__heading,
          headingProps.className
        )}
      >
        <Toggler
          className={classNames(
            'accordion__toggle',
            styles.accordion__toggle,
            isCollapsed && [
              'accordion__toggle--checked',
              styles.accordion__toggle_collapsed
            ]
          )}
          heading={heading}
          onToggle={() => setCollapsed((p) => !p)}
        />
      </div>
      <div
        {...contentProps}
        aria-hidden={isCollapsed}
        className={classNames(
          'accordion__content',
          styles.accordion__content,
          contentProps.className,
          isCollapsed && [
            'accordion__content--collapsed',
            styles.accordion__content_collapsed
          ]
        )}
      >
        {typeof children !== 'function' ? children : children(isCollapsed)}
      </div>
    </div>
  );
}

Accordion.propTypes = {
  heading: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.node
  ]).isRequired,
  defaultIsCollapsed: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.node
  ]),
  onToggle: PropTypes.func
};

export default Accordion;
