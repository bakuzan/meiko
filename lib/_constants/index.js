export { default as Strings } from './strings';
export { default as Types } from './types';
export { default as Urls } from './urls';
export { default as Icons } from './icons';
export { PositionEnum, KeyCodes, OPEN_KEYS, CLOSE_KEYS } from './enums';
