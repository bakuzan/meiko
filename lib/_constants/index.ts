import _Strings from './strings';
import _Types from './types';
import _Urls from './urls';
import _Icons from './icons';
import * as _Enums from './enums';

export const Strings = _Strings;
export const Types = _Types;
export const Urls = _Urls;
export const Icons = _Icons;
export const Enums = _Enums;
