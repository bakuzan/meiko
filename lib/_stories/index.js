import React from 'react';
// import { storiesOf } from '@storybook/react';

import './styles/index.scss';

// storiesOf('Index', module).add('basic', () => <div>Exists to load styles</div>);

export const storyStyle = {
  maxWidth: '300px',
  margin: 'auto',
  marginTop: '20px'
};

const styles = { display: 'flex', justifyContent: 'center', marginTop: '30px' };
export const CenterDecorator = (storyFn) => (
  <div style={styles}>{storyFn()}</div>
);
