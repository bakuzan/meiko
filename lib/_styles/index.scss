@import './partials/variables.scss';
@import './box-model.scss';
@import './list.scss';
@import './input-controls.scss';

.button-group {
  display: flex;
  justify-content: space-around;
  align-items: center;
  padding: 5px;
  margin: 5px 0;

  &.centered {
    justify-content: center;
  }
  &.right-aligned {
    justify-content: flex-end;
  }
}
