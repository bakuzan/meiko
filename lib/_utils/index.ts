import adjustDateMonth from 'ayaka/adjustDateMonth';
import adjustDateYear from 'ayaka/adjustDateYear';
import { capitalise } from 'ayaka/capitalise';
import checkIfDatePartsMatch from 'ayaka/checkIfDatePartsMatch';
import convertToBase64 from 'ayaka/convertToBase64';
import createListeners from 'ayaka/createListeners';
import debounce from 'ayaka/debounce';
import formatDateForInput from 'ayaka/formatDateForInput';
import generateUniqueId from 'ayaka/generateUniqueId';
import getDaysInMonthForDate from 'ayaka/getDaysInMonthForDate';
import getDifferenceFromMonday from 'ayaka/daysDifferenceFromMonday';
import getDifferenceFromSunday from 'ayaka/daysDifferenceFromSunday';
import getElementCoordinates from 'ayaka/getElementCoordinates';
import getEventValue from 'ayaka/getEventValue';
import getFirstDateOfMonth from 'ayaka/getFirstDateOfMonth';
import getLastDateOfMonth from 'ayaka/getLastDateOfMonth';
import getMonthName from 'ayaka/getMonthName';
import { getTimeoutSeconds } from 'ayaka/getTimeout';
import isArray from 'ayaka/isArray';
import { isBefore, isBeforeOrEqual } from 'ayaka/isBefore';
import isNumber from 'ayaka/isNumber';
import isObject from 'ayaka/isObject';
import isString from 'ayaka/isString';
import { lighten, darken } from 'ayaka/palette';

export {
  adjustDateMonth,
  adjustDateYear,
  capitalise,
  convertToBase64,
  createListeners,
  debounce,
  generateUniqueId,
  getElementCoordinates,
  getEventValue,
  getTimeoutSeconds,
  isArray,
  isNumber,
  isObject,
  isString,
  checkIfDatePartsMatch,
  formatDateForInput,
  getMonthName,
  getDaysInMonthForDate,
  getFirstDateOfMonth,
  getLastDateOfMonth,
  getDifferenceFromMonday,
  getDifferenceFromSunday,
  isBefore,
  isBeforeOrEqual,
  lighten,
  darken
};
