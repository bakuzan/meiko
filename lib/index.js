// Components
export { default as Alert } from './Alert';
export { default as AppInformation } from './AppInformation';
export { default as AutocompleteInput } from './AutocompleteInput';
export { Button, withButtonisation } from './Button';
export { default as ChipListInput } from './ChipListInput';
export { default as ClearableInput } from './ClearableInput';
export { default as DateSelector } from './DateSelector';
export { default as Dialog } from './Dialog';
export { default as DropdownMenu } from './DropdownMenu';
export { default as FileUploader } from './FileUploader';
export { default as Form } from './Form';
export { default as FormControls } from './FormControls';
export { default as Grid } from './Grid';
export { default as Header } from './Header';
export { default as Image } from './Image';
export { default as ImageSelector } from './ImageSelector';
export { default as List } from './List';
export { SimpleLoading, Loading } from './Loadable';
export { default as LoadableContent } from './LoadableContent';
export { default as LoadingBouncer } from './LoadingBouncer';
export { default as LoadingSpinner } from './LoadingSpinner';
export { default as Logo } from './Logo';

export { default as MultiSelect } from './MultiSelect';
export { default as RadioButton } from './RadioButton';
export { default as RadioToggle } from './RadioToggle';
export { default as RatingControl } from './RatingControl';
export { default as RequestIndicator } from './RequestIndicator';
export { default as ScrollTopButton } from './ScrollTopButton';
export { default as SelectBox } from './SelectBox';
export { default as Sidebar } from './Sidebar';
export { default as Tabs } from './Tabs';
export { default as TagCloudSelector } from './TagCloudSelector';
export { default as Tickbox } from './Tickbox';
export { default as Toaster } from './Toaster';
export { default as Tooltip } from './Tooltip';

export { default as NewTabLink } from './NewTabLink';
export { default as Portal } from './Portal';

// Hooks
export { useGlobalStyles } from './hooks/useGlobalStyles';

// Utils
export { default as toaster, ToasterService } from './utils/toasterService';
export { default as addOutsideClick } from './utils/addOutsideClick';

export {
  PositionEnum,
  KeyCodes,
  OPEN_KEYS,
  CLOSE_KEYS,
  Icons,
  Strings,
  Types,
  Urls
} from './constants';
