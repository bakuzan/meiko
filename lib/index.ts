import './styles/index.scss';

import * as Components from './components';
import { default as fetchFromServer, handleErrorResponse } from './utils/fetch';
import * as Constants from './constants';

export default {
  ...Components,
  Constants,
  fetchFromServer,
  handleErrorResponse
};
