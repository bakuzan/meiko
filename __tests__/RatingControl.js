import React from 'react';

import { RatingControl } from '../lib';

const mockedChangeFn = jest.fn();

it('should render with minimum props', function () {
  const { container } = render(<RatingControl id="jest" value={0} />);

  expect(container.firstChild).toBeTruthy();
  expect(container).toMatchSnapshot();
});

it('should call onChange', async function () {
  const { container, getByLabelText } = render(
    <RatingControl id="jest" value={0} onChange={mockedChangeFn} />
  );

  const user = userEvent.setup();
  await user.click(getByLabelText('4/10'));

  expect(mockedChangeFn).toHaveBeenCalled();
  expect(container).toMatchSnapshot();
});

it('should return value of 0 if active value is selected', async function () {
  const { container, getByLabelText } = render(
    <RatingControl id="jest" value={5} onChange={mockedChangeFn} />
  );

  const user = userEvent.setup();
  await user.click(getByLabelText('5/10 (selected)'));

  expect(mockedChangeFn).toHaveBeenCalledWith({
    target: {
      type: 'radio',
      value: 0,
      name: undefined
    }
  });
  expect(container).toMatchSnapshot();
});

it('should render "maxRating" number of inputs', function () {
  const { container, rerender } = render(
    <RatingControl id="jest" value={0} onChange={mockedChangeFn} />
  );

  // Default prop, plus 1 for the label
  expect(container.firstChild.children.length).toEqual(10 + 1);

  rerender(
    <RatingControl
      id="jest"
      value={0}
      maxRating={3}
      onChange={mockedChangeFn}
    />
  );
  expect(container.firstChild.children.length).toEqual(3 + 1);
  expect(container).toMatchSnapshot();

  rerender(
    <RatingControl
      id="jest"
      value={0}
      maxRating={15}
      onChange={mockedChangeFn}
    />
  );
  expect(container.firstChild.children.length).toEqual(15 + 1);
  expect(container).toMatchSnapshot();
});
