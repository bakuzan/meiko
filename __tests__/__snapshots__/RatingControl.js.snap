// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call onChange 1`] = `
<div>
  <div
    class="rating-control  mko_1a3l7p3-control"
    role="radiogroup"
  >
    <label
      aria-label="1/10"
      class="rating-control__option rating-control__option--low  mko_1a3l7p3-control__option  mko_1a3l7p3-low"
      for="jest-1"
      title="1/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-1"
        type="radio"
        value="1"
      />
    </label>
    <label
      aria-label="2/10"
      class="rating-control__option rating-control__option--low  mko_1a3l7p3-control__option  mko_1a3l7p3-low"
      for="jest-2"
      title="2/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-2"
        type="radio"
        value="2"
      />
    </label>
    <label
      aria-label="3/10"
      class="rating-control__option rating-control__option--low  mko_1a3l7p3-control__option  mko_1a3l7p3-low"
      for="jest-3"
      title="3/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-3"
        type="radio"
        value="3"
      />
    </label>
    <label
      aria-label="4/10"
      class="rating-control__option rating-control__option--average  mko_1a3l7p3-control__option  mko_1a3l7p3-average"
      for="jest-4"
      title="4/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-4"
        type="radio"
        value="4"
      />
    </label>
    <label
      aria-label="5/10"
      class="rating-control__option rating-control__option--average  mko_1a3l7p3-control__option  mko_1a3l7p3-average"
      for="jest-5"
      title="5/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-5"
        type="radio"
        value="5"
      />
    </label>
    <label
      aria-label="6/10"
      class="rating-control__option rating-control__option--average  mko_1a3l7p3-control__option  mko_1a3l7p3-average"
      for="jest-6"
      title="6/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-6"
        type="radio"
        value="6"
      />
    </label>
    <label
      aria-label="7/10"
      class="rating-control__option rating-control__option--good  mko_1a3l7p3-control__option  mko_1a3l7p3-good"
      for="jest-7"
      title="7/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-7"
        type="radio"
        value="7"
      />
    </label>
    <label
      aria-label="8/10"
      class="rating-control__option rating-control__option--good  mko_1a3l7p3-control__option  mko_1a3l7p3-good"
      for="jest-8"
      title="8/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-8"
        type="radio"
        value="8"
      />
    </label>
    <label
      aria-label="9/10"
      class="rating-control__option rating-control__option--great  mko_1a3l7p3-control__option  mko_1a3l7p3-great"
      for="jest-9"
      title="9/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-9"
        type="radio"
        value="9"
      />
    </label>
    <label
      aria-label="10/10"
      class="rating-control__option rating-control__option--great  mko_1a3l7p3-control__option  mko_1a3l7p3-great"
      for="jest-10"
      title="10/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-10"
        type="radio"
        value="10"
      />
    </label>
    <div
      class="rating-control__label  mko_1a3l7p3-control__label"
    />
  </div>
</div>
`;

exports[`should render "maxRating" number of inputs 1`] = `
<div>
  <div
    class="rating-control  mko_1a3l7p3-control"
    role="radiogroup"
  >
    <label
      aria-label="1/3"
      class="rating-control__option rating-control__option--low  mko_1a3l7p3-control__option  mko_1a3l7p3-low"
      for="jest-1"
      title="1/3"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-1"
        type="radio"
        value="1"
      />
    </label>
    <label
      aria-label="2/3"
      class="rating-control__option rating-control__option--low  mko_1a3l7p3-control__option  mko_1a3l7p3-low"
      for="jest-2"
      title="2/3"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-2"
        type="radio"
        value="2"
      />
    </label>
    <label
      aria-label="3/3"
      class="rating-control__option rating-control__option--low  mko_1a3l7p3-control__option  mko_1a3l7p3-low"
      for="jest-3"
      title="3/3"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-3"
        type="radio"
        value="3"
      />
    </label>
    <div
      class="rating-control__label  mko_1a3l7p3-control__label"
    />
  </div>
</div>
`;

exports[`should render "maxRating" number of inputs 2`] = `
<div>
  <div
    class="rating-control  mko_1a3l7p3-control"
    role="radiogroup"
  >
    <label
      aria-label="1/15"
      class="rating-control__option rating-control__option--low  mko_1a3l7p3-control__option  mko_1a3l7p3-low"
      for="jest-1"
      title="1/15"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-1"
        type="radio"
        value="1"
      />
    </label>
    <label
      aria-label="2/15"
      class="rating-control__option rating-control__option--low  mko_1a3l7p3-control__option  mko_1a3l7p3-low"
      for="jest-2"
      title="2/15"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-2"
        type="radio"
        value="2"
      />
    </label>
    <label
      aria-label="3/15"
      class="rating-control__option rating-control__option--low  mko_1a3l7p3-control__option  mko_1a3l7p3-low"
      for="jest-3"
      title="3/15"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-3"
        type="radio"
        value="3"
      />
    </label>
    <label
      aria-label="4/15"
      class="rating-control__option rating-control__option--average  mko_1a3l7p3-control__option  mko_1a3l7p3-average"
      for="jest-4"
      title="4/15"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-4"
        type="radio"
        value="4"
      />
    </label>
    <label
      aria-label="5/15"
      class="rating-control__option rating-control__option--average  mko_1a3l7p3-control__option  mko_1a3l7p3-average"
      for="jest-5"
      title="5/15"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-5"
        type="radio"
        value="5"
      />
    </label>
    <label
      aria-label="6/15"
      class="rating-control__option rating-control__option--average  mko_1a3l7p3-control__option  mko_1a3l7p3-average"
      for="jest-6"
      title="6/15"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-6"
        type="radio"
        value="6"
      />
    </label>
    <label
      aria-label="7/15"
      class="rating-control__option rating-control__option--good  mko_1a3l7p3-control__option  mko_1a3l7p3-good"
      for="jest-7"
      title="7/15"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-7"
        type="radio"
        value="7"
      />
    </label>
    <label
      aria-label="8/15"
      class="rating-control__option rating-control__option--good  mko_1a3l7p3-control__option  mko_1a3l7p3-good"
      for="jest-8"
      title="8/15"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-8"
        type="radio"
        value="8"
      />
    </label>
    <label
      aria-label="9/15"
      class="rating-control__option rating-control__option--great  mko_1a3l7p3-control__option  mko_1a3l7p3-great"
      for="jest-9"
      title="9/15"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-9"
        type="radio"
        value="9"
      />
    </label>
    <label
      aria-label="10/15"
      class="rating-control__option rating-control__option--great  mko_1a3l7p3-control__option  mko_1a3l7p3-great"
      for="jest-10"
      title="10/15"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-10"
        type="radio"
        value="10"
      />
    </label>
    <label
      aria-label="11/15"
      class="rating-control__option rating-control__option--great  mko_1a3l7p3-control__option  mko_1a3l7p3-great"
      for="jest-11"
      title="11/15"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-11"
        type="radio"
        value="11"
      />
    </label>
    <label
      aria-label="12/15"
      class="rating-control__option rating-control__option--great  mko_1a3l7p3-control__option  mko_1a3l7p3-great"
      for="jest-12"
      title="12/15"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-12"
        type="radio"
        value="12"
      />
    </label>
    <label
      aria-label="13/15"
      class="rating-control__option rating-control__option--great  mko_1a3l7p3-control__option  mko_1a3l7p3-great"
      for="jest-13"
      title="13/15"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-13"
        type="radio"
        value="13"
      />
    </label>
    <label
      aria-label="14/15"
      class="rating-control__option rating-control__option--great  mko_1a3l7p3-control__option  mko_1a3l7p3-great"
      for="jest-14"
      title="14/15"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-14"
        type="radio"
        value="14"
      />
    </label>
    <label
      aria-label="15/15"
      class="rating-control__option rating-control__option--great  mko_1a3l7p3-control__option  mko_1a3l7p3-great"
      for="jest-15"
      title="15/15"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-15"
        type="radio"
        value="15"
      />
    </label>
    <div
      class="rating-control__label  mko_1a3l7p3-control__label"
    />
  </div>
</div>
`;

exports[`should render with minimum props 1`] = `
<div>
  <div
    class="rating-control rating-control--read-only  mko_1a3l7p3-control  mko_1a3l7p3-control_readOnly"
    role="radiogroup"
  >
    <label
      aria-label="1/10"
      class="rating-control__option rating-control__option--low  mko_1a3l7p3-control__option  mko_1a3l7p3-low"
      for="jest-1"
      title="1/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-1"
        type="radio"
        value="1"
      />
    </label>
    <label
      aria-label="2/10"
      class="rating-control__option rating-control__option--low  mko_1a3l7p3-control__option  mko_1a3l7p3-low"
      for="jest-2"
      title="2/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-2"
        type="radio"
        value="2"
      />
    </label>
    <label
      aria-label="3/10"
      class="rating-control__option rating-control__option--low  mko_1a3l7p3-control__option  mko_1a3l7p3-low"
      for="jest-3"
      title="3/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-3"
        type="radio"
        value="3"
      />
    </label>
    <label
      aria-label="4/10"
      class="rating-control__option rating-control__option--average  mko_1a3l7p3-control__option  mko_1a3l7p3-average"
      for="jest-4"
      title="4/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-4"
        type="radio"
        value="4"
      />
    </label>
    <label
      aria-label="5/10"
      class="rating-control__option rating-control__option--average  mko_1a3l7p3-control__option  mko_1a3l7p3-average"
      for="jest-5"
      title="5/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-5"
        type="radio"
        value="5"
      />
    </label>
    <label
      aria-label="6/10"
      class="rating-control__option rating-control__option--average  mko_1a3l7p3-control__option  mko_1a3l7p3-average"
      for="jest-6"
      title="6/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-6"
        type="radio"
        value="6"
      />
    </label>
    <label
      aria-label="7/10"
      class="rating-control__option rating-control__option--good  mko_1a3l7p3-control__option  mko_1a3l7p3-good"
      for="jest-7"
      title="7/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-7"
        type="radio"
        value="7"
      />
    </label>
    <label
      aria-label="8/10"
      class="rating-control__option rating-control__option--good  mko_1a3l7p3-control__option  mko_1a3l7p3-good"
      for="jest-8"
      title="8/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-8"
        type="radio"
        value="8"
      />
    </label>
    <label
      aria-label="9/10"
      class="rating-control__option rating-control__option--great  mko_1a3l7p3-control__option  mko_1a3l7p3-great"
      for="jest-9"
      title="9/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-9"
        type="radio"
        value="9"
      />
    </label>
    <label
      aria-label="10/10"
      class="rating-control__option rating-control__option--great  mko_1a3l7p3-control__option  mko_1a3l7p3-great"
      for="jest-10"
      title="10/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-10"
        type="radio"
        value="10"
      />
    </label>
    <div
      class="rating-control__label  mko_1a3l7p3-control__label"
    />
  </div>
</div>
`;

exports[`should return value of 0 if active value is selected 1`] = `
<div>
  <div
    class="rating-control  mko_1a3l7p3-control"
    role="radiogroup"
  >
    <label
      aria-label="1/10"
      class="rating-control__option rating-control__option--low rating-control__option--past  mko_1a3l7p3-control__option  mko_1a3l7p3-low  mko_1a3l7p3-control__option_past"
      for="jest-1"
      title="1/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input  mko_1a3l7p3-control__input_highlighted"
        id="jest-1"
        type="radio"
        value="1"
      />
    </label>
    <label
      aria-label="2/10"
      class="rating-control__option rating-control__option--low rating-control__option--past  mko_1a3l7p3-control__option  mko_1a3l7p3-low  mko_1a3l7p3-control__option_past"
      for="jest-2"
      title="2/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input  mko_1a3l7p3-control__input_highlighted"
        id="jest-2"
        type="radio"
        value="2"
      />
    </label>
    <label
      aria-label="3/10"
      class="rating-control__option rating-control__option--low rating-control__option--past  mko_1a3l7p3-control__option  mko_1a3l7p3-low  mko_1a3l7p3-control__option_past"
      for="jest-3"
      title="3/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input  mko_1a3l7p3-control__input_highlighted"
        id="jest-3"
        type="radio"
        value="3"
      />
    </label>
    <label
      aria-label="4/10"
      class="rating-control__option rating-control__option--average rating-control__option--past  mko_1a3l7p3-control__option  mko_1a3l7p3-average  mko_1a3l7p3-control__option_past"
      for="jest-4"
      title="4/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input  mko_1a3l7p3-control__input_highlighted"
        id="jest-4"
        type="radio"
        value="4"
      />
    </label>
    <label
      aria-label="5/10 (selected)"
      class="rating-control__option rating-control__option--average rating-control__option--selected  mko_1a3l7p3-control__option  mko_1a3l7p3-average  mko_1a3l7p3-control__option_selected"
      for="jest-5"
      title="5/10 (selected)"
    >
      <input
        checked=""
        class="rating-control__input  mko_1a3l7p3-control__input  mko_1a3l7p3-control__input_highlighted"
        id="jest-5"
        type="radio"
        value="5"
      />
    </label>
    <label
      aria-label="6/10"
      class="rating-control__option rating-control__option--average  mko_1a3l7p3-control__option  mko_1a3l7p3-average"
      for="jest-6"
      title="6/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-6"
        type="radio"
        value="6"
      />
    </label>
    <label
      aria-label="7/10"
      class="rating-control__option rating-control__option--good  mko_1a3l7p3-control__option  mko_1a3l7p3-good"
      for="jest-7"
      title="7/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-7"
        type="radio"
        value="7"
      />
    </label>
    <label
      aria-label="8/10"
      class="rating-control__option rating-control__option--good  mko_1a3l7p3-control__option  mko_1a3l7p3-good"
      for="jest-8"
      title="8/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-8"
        type="radio"
        value="8"
      />
    </label>
    <label
      aria-label="9/10"
      class="rating-control__option rating-control__option--great  mko_1a3l7p3-control__option  mko_1a3l7p3-great"
      for="jest-9"
      title="9/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-9"
        type="radio"
        value="9"
      />
    </label>
    <label
      aria-label="10/10"
      class="rating-control__option rating-control__option--great  mko_1a3l7p3-control__option  mko_1a3l7p3-great"
      for="jest-10"
      title="10/10"
    >
      <input
        class="rating-control__input  mko_1a3l7p3-control__input"
        id="jest-10"
        type="radio"
        value="10"
      />
    </label>
    <div
      class="rating-control__label  mko_1a3l7p3-control__label"
    />
  </div>
</div>
`;
